File: JAVA/RemoveElement.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.util.Arrays;

public class RemoveElement {

    public static void main(String[] args) {
        System.out.printf("Resultado: %d\n", removeElement( new int[]{3,2,2,3}, 3 ));
    }

    public static int removeElement(int[] nums, int val) {
        int count = 0;

        for ( int i=0; i < nums.length; i++ ) {
            if ( nums[i] == val ) nums[i] = '_'; else count++;
        }
        Arrays.sort( nums );

        return count;
    }

}

File: GO/RemoveElement.go
/**
 * @version GO 1.19.4
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import "sort"

func removeElement(nums []int, val int) int {
	total := 0

	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			total++
		} else {
			nums[i] = '_'
		}
	}
	sort.Ints(nums)

	return total
}
